### PR TITLE
Chart improvements

### PR DESCRIPTION
**Pull request by peuter**
* show grid lines by default and add confuguration option to select fpr wich axis they should be shown
* add bar charts
* add stacked-bar charts
* add confugurable curve interpolations for charts

refs #1299 

Also in this PR:
* option to change colors of the round-progress bar
* fix layout problem when excluding tile, if its a widget the parent must be exluded

### CODE REVIEW

File: source/class/cv/ui/structure/tile/components/Chart.js [hunk 1]
```diff
@@ -656,17 +682,46 @@ qx.Class.define('cv.ui.structure.tile.components.Chart', {
...
+
+              case 'natural':
+                curveFunction = d3.curveNatural;
+                break;
+            }
+            if (curveFunction) {
+              // Construct a line generator.
+              lineFunctions[curveName] = d3
+                .line()
+                //.defined(i => D[i])
```
Comment (ChristianMayer) [inline source/class/cv/ui/structure/tile/components/Chart.js:718]: Is the comment here on purpose or just a left over missed to delete?

File: source/class/cv/ui/structure/tile/components/Chart.js [hunk 2]
```diff
@@ -681,29 +736,60 @@ qx.Class.define('cv.ui.structure.tile.components.Chart', {
...
+
+              case 'natural':
+                curveFunction = d3.curveNatural;
+                break;
+            }
+            if (curveFunction) {
+              // Construct a line generator.
+              areaFunctions[curveName] = d3
+                .area()
+                //.defined(i => D[i])
```
Comment (ChristianMayer) [inline source/class/cv/ui/structure/tile/components/Chart.js:770]: Is the comment here on purpose or just a left over missed to delete?

File: source/class/cv/ui/structure/tile/components/RoundProgress.js [hunk 3]
```diff
@@ -107,40 +107,69 @@ qx.Class.define('cv.ui.structure.tile.components.RoundProgress', {
       }
       const type = this.getType();
       let height = type === 'semiCircle' ? radius : radius * 2;
-      let code = `<svg height="${height}" width="${radius * 2}" type="${type}">`;
+      const ns = 'http://www.w3.org/2000/svg';
+      const svg = document.createElementNS(ns, 'svg');
+      svg.setAttribute('height', height);
+      svg.setAttribute('width', '' + (radius * 2));
+      svg.setAttribute('type', type);
+      //let code = `<svg height="${height}" width="${radius * 2}" type="${type}">`;
```
Comment (ChristianMayer) [inline source/class/cv/ui/structure/tile/components/RoundProgress.js:115]: Is the comment here on purpose or just a left over missed to delete?

**Review by peuter — COMMENTED** [review]
remove leftovers

File: source/class/cv/ui/structure/tile/components/Chart.js (hunk 2 above)
Comment (peuter) [reply source/class/cv/ui/structure/tile/components/Chart.js]: Those are all leftovers.

**Review by ChristianMayer — APPROVED** [review]

**Pull request merged**